clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -1875,7 +1907,9 @@ private void completeUnsubscribeApplicationEventSuccessfully() {
     private void completeAssignmentChangeEventSuccessfully() {
         doAnswer(invocation -> {
             AssignmentChangeEvent event = invocation.getArgument(0);
-            consumer.subscriptions().assignFromUser(new HashSet<>(event.partitions()));
+            HashSet<TopicPartition> partitions = new HashSet<>(event.partitions());
+            consumer.subscriptions().assignFromUser(partitions);
+            consumer.setGroupAssignmentSnapshot(partitions);
```

Comment:
this func if about manual assignment so I would expect we don't need to set the group assignment snapshot right?

Reply:
You're right. That's a holdover from when I was updating the snapshot for manually-assigned partitions too.